lib/model.js

```diff
@@ -3247,7 +3247,8 @@ class Model {
       // If not raw, and there's a custom setter
       if (!options.raw && this._customSetters[key]) {
         this._customSetters[key].call(this, value, key);
-        if (!Utils.isPrimitive(value) && value !== null || value !== originalValue) {
+        const newValue = this.dataValues[key];
```

Comment:
Add a comment saying "custom setter should have changed value, get that changed value"

Also add a comment saying "TODO: v5 make setters return new value instead of changing internal store"